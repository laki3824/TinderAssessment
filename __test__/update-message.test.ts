const getRecentMessageFromChannel = require( "./common-utils" );
const { WebClient } = require('@slack/web-api');
const dotenv = require('dotenv')

describe('Update messages tests - ', () => {

    beforeAll(() => {
        console.log("Start tests....");
        dotenv.config();
    });
  
    afterAll(() => {
      console.log("End tests....")
    });

    /**
     * Edit a posted message from a channel.
     * https://api.slack.com/methods/chat.update
     */
    test('Edit a text message using a valid messageId', async() => {

        //Post a message before trying to edit one, so this test can be executed independently
        const client = new WebClient(process.env.USER_API_KEY);

        const channelId = process.env.CHANNEL_ID;

        const response = await client.chat.postMessage({
            channel: process.env.CHANNEL_NAME,
            text: "This is a text message",
        });

        //Verify whether the message was posted to the channel
        expect(response.ok).toBe(true);

        //Retrieve the recent message from the channel
        const message = await getRecentMessageFromChannel(client, channelId);

        const messageId = message.ts;

        console.log("Retrieved messageId: " + messageId);
        
        //Delete the message using the messageId/ts
        const editResponse = await client.chat.update({
            channel: channelId,
            ts: messageId,
            text: "This is an updated text message",
        });

        //Verify the ts/messageId is not modified with this update
        expect(editResponse.ts).toEqual(messageId);

        //Validate the response snap shot
        expect(editResponse).toMatchSnapshot({
            ts: expect.any(String)
        });

    });

});
