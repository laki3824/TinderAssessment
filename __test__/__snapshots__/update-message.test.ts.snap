// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Update messages tests -  Edit a text message using a valid messageId 1`] = `
Object {
  "channel": "C029Y3JNWTS",
  "message": Object {
    "bot_id": "B02995EBYP4",
    "bot_profile": Object {
      "app_id": "A029937T49G",
      "deleted": false,
      "icons": Object {
        "image_36": "https://a.slack-edge.com/80588/img/plugins/app/bot_36.png",
        "image_48": "https://a.slack-edge.com/80588/img/plugins/app/bot_48.png",
        "image_72": "https://a.slack-edge.com/80588/img/plugins/app/service_72.png",
      },
      "id": "B02995EBYP4",
      "name": "Demo App",
      "team_id": "T029Y3JN508",
      "updated": 1627364491,
    },
    "team": "T029Y3JN508",
    "text": "This is an updated text message",
    "type": "message",
    "user": "U0298CLJHH8",
  },
  "ok": true,
  "response_metadata": Object {
    "acceptedScopes": Array [
      "chat:write",
    ],
    "scopes": Array [
      "identify",
      "channels:history",
      "chat:write",
    ],
  },
  "text": "This is an updated text message",
  "ts": Any<String>,
}
`;
